sklearn/utils/tests/test_validation.py

```diff
@@ -417,6 +418,11 @@ def test_check_array_dtype_warning():
     assert_equal(X_checked.format, 'csr')
 
 
+def test_check_array_warn_on_dtype_deprecation():
+    X = np.asarray([[0.0], [1.0]])
+    assert_warns(DeprecationWarning, check_array, X, warn_on_dtype=True)
```

Comment:
Please use pytest.wars with the `match` parameter as shown [here](https://docs.pytest.org/en/latest/warnings.html#warns), instead of assert_warns